File: JavaKotlin/app/src/main/java/com/example/javakotlin/MainActivity.java
package com.example.javakotlin;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button boton1 = findViewById(R.id.boton1);
        Button boton2 = findViewById(R.id.boton2);

        boton1.setOnClickListener(v -> {
                    Intent intent = new Intent(MainActivity.this, MainActivityKotlin.class);
                    startActivity(intent);
                    Toast.makeText(MainActivity.this,"Pinchando en el botón",Toast.LENGTH_SHORT).show();
                }
        );

        boton2.setOnClickListener(v -> {
                    Intent intent = new Intent(MainActivity.this, MainActivityKotlinOtra.class);
                    startActivity(intent);
                    Toast.makeText(MainActivity.this,"Pinchando en el botón",Toast.LENGTH_SHORT).show();
                }
        );


        TextView tv1 = findViewById(R.id.tv1);

        tv1.setOnClickListener(
                v-> {
                    Toast.makeText(MainActivity.this,"Pinchando en el texto",Toast.LENGTH_SHORT).show();
                }
        );
    }
}

File: EjercicioEventosKotlin1/app/src/main/java/com/example/ejercicioeventoskotlin1/MainActivity.kt
package com.example.ejercicioeventoskotlin1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

import com.example.ejercicioeventoskotlin1.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        var contadorBoton1=0
        var contadorBoton2=0
        var total=0

        binding.button1.setText("Contador A ($contadorBoton1)")
        binding.button2.setText("Contador B ($contadorBoton2)")

        binding.button1.setOnClickListener{
            contadorBoton1++
            total++
            binding.textView1.setText("$contadorBoton1")
            (it as Button).setText("Contador A ($contadorBoton1)")
            binding.textView3.setText("$total")
            Toast.makeText(this@MainActivity,"$contadorBoton1", Toast.LENGTH_SHORT).show()

        }

        binding.button2.setOnClickListener{
            contadorBoton2++
            total++
            binding.textView2.setText("$contadorBoton2")
            (it as Button).setText("Contador B ($contadorBoton2)")
            binding.textView3.setText("$total")
            Toast.makeText(this@MainActivity,"$contadorBoton2",Toast.LENGTH_SHORT).show()
        }



    }
}